AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/src/DefaultKeyStorageInterface.dfy

```diff
@@ -585,78 +585,83 @@ module DefaultKeyStorageInterface {
       ensures ValidState()
       ensures GetMutationEnsuresPublicly(input, output)
       ensures unchanged(History)
-
-      ensures |ddbClient.History.TransactGetItems| == |old(ddbClient.History.TransactGetItems)| + 1
       ensures output.Success?
               ==>
-                && Seq.Last(ddbClient.History.TransactGetItems).output.Success?
-
+                |ddbClient.History.GetItem| == |old(ddbClient.History.GetItem)| + 2
+      ensures output.Success?
+              ==>
+                && Seq.Last(ddbClient.History.GetItem).output.Success?
```

Comment:
this will only ensure that only the last getItem succeeds - not sure that's what you want

Reply:
Yes, we could radically improve the Formal Verification of this method. I just am not sure it is worth it. I do not think formal verification is normally used as unit tests. We have many, many, integration tests that prove this method works.